src/tests/daos_mw_fi.c

```diff
@@ -90,22 +106,29 @@ fi_dfs(daos_handle_t poh, daos_handle_t coh, const char *op, const char *path, d
 		return -DER_INVAL;
 	}
 
-	D_STRNDUP(dir, path, PATH_MAX);
-	if (dir == NULL)
-		return -DER_NOMEM;
-	D_STRNDUP(file, path, PATH_MAX);
-	if (file == NULL)
-		D_GOTO(out_path, rc = -DER_NOMEM);
-
 	rc = dfs_mount(poh, coh, O_RDWR, &dfs);
 	if (rc) {
 		fprintf(stderr, "dfs_mount() failed: (%d)\n", rc);
 		D_GOTO(out_path, rc);
 	}
 
-	fname = basename(file);
-	dirp = dirname(dir);
-	printf("punching %s from %s\n", fname, dirp);
+	if (op == PUNCH_ENTRY) {
+		D_STRNDUP(dir, path, PATH_MAX);
+		if (dir == NULL)
+			return -DER_NOMEM;
+		D_STRNDUP(file, path, PATH_MAX);
+		if (file == NULL)
+			D_GOTO(out_path, rc = -DER_NOMEM);
```

Comment:
Shouldn't both of these `goto out_dfs` so dfs is unmounted?

Reply:
ah yes.. i moved them but forgot to update the goto. thx